File: java/05_048.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class Node {
        public int val;
        public Node left;
        public Node right;
        
        public Node(int val) {
            this.val = val;
        }
    }
    
    public int maxDepth(Node root) {
        if (root == null) {
            return 0;
        }
        return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
    }
}

File: data structures and algorithms/cs/05_048.cs
using System;
using System.Collections.Generic;

public class MainClass
{
   public static void Main(string[] args)
   {
      // Main method code goes here
   }

   public class Node
   {
      public int Val { get; set; }
      public Node Left { get; set; }
      public Node Right { get; set; }

      public Node(int val)
      {
         Val = val;
      }
   }

   public int MaxDepth(Node root)
   {
      if (root == null)
      {
         return 0;
      }
      return 1 + Math.Max(MaxDepth(root.Left), MaxDepth(root.Right));
   }
}
